modules/swagger-codegen/src/main/resources/elixir/deserializer.ex.mustache

```diff
@@ -21,11 +21,15 @@ defmodule {{moduleName}}.Deserializer do
     |> Map.update!(field, &(Map.new(&1, fn {key, val} -> {key, Poison.Decode.decode(val, Keyword.merge(options, [as: struct(mod)]))} end)))
   end
   def deserialize(model, field, :date, _, _options) do
-    case DateTime.from_iso8601(Map.get(model, field)) do
-      {:ok, datetime} ->
-        Map.put(model, field, datetime)
-      _ ->
-        model
+    value = Map.get(model, field)
+    case is_binary(value) do
+      true -> case DateTime.from_iso8601(value) do
+                {:ok, datetime} ->
```

Comment:
Another bug is that [`DateTime.from_iso8601`](https://hexdocs.pm/elixir/DateTime.html#from_iso8601/2) returns a tuple of size 3.

So it should actually be `{:ok, datetime, _offset}`

Reply:
Latest commit should fix this, would that be ok?